code/game/objects/items/weapons/batons.dm

```diff
@@ -71,15 +73,21 @@
 		user.visible_message("<span class='danger'>[user] pulses [target]'s sensors with [src]!</span>",\
 							"<span class='danger'>You pulse [target]'s sensors with [src]!</span>")
 		on_silicon_stun(target, user)
-	else
-		// Check for shield/countering
-		if(ishuman(target))
-			var/mob/living/carbon/human/H = target
-			if(H.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
-				return FALSE
+
+	// Check for shield/countering
+	if(ishuman(target))
```

Comment:
make this if and the one below it `else if`s please, that will make sure that we can skip two istype checks every time we pass a silicon through here